modules/series-service-impl/src/main/java/org/opencastproject/series/impl/SeriesServiceImpl.java

```diff
@@ -504,53 +504,78 @@ public void repopulate() throws IndexRebuildException {
+                      try {
+                        catalog = DublinCoreXmlFormat.read(series.getDublinCoreXML());
+                        updateFunctions.add(getMetadataUpdateFunction(seriesId, catalog, organization.getId()));
+                      } catch (IOException | ParserConfigurationException | SAXException e) {
+                        logger.error("Could not read dublincore XML of series {}.", seriesId, e);
+                        return;
+                      }
+
+                      String aclStr = series.getAccessControl();
+                      if (StringUtils.isNotBlank(aclStr)) {
+                        try {
+                          AccessControlList acl = AccessControlParser.parseAcl(aclStr);
+                          updateFunctions.add(getAclUpdateFunction(seriesId, acl, organization.getId()));
+                        } catch (Exception ex) {
+                          logger.error("Unable to parse ACL of series {}.", seriesId, ex);
+                        }
+                      }
+
+                      try {
+                        Map<String, String> properties = persistence.getSeriesProperties(seriesId);
+                        updateFunctions.add(getThemePropertyUpdateFunction(seriesId,
+                                Optional.ofNullable(properties.get(THEME_PROPERTY_NAME)), organization.getId()));
+                      } catch (NotFoundException | SeriesServiceDatabaseException e) {
+                        logger.error("Error reading properties of series {}", seriesId, e);
+                      }
+                      updateFunctionBundle.add(updateFunctions.toArray(new Function[0]));
                     }
-                  }
-
-                  try {
-                    Map<String, String> properties = persistence.getSeriesProperties(seriesId);
-                    updateFunctions.add(getThemePropertyUpdateFunction(seriesId,
-                            Optional.ofNullable(properties.get(THEME_PROPERTY_NAME)), organization.getId()));
-                  } catch (NotFoundException | SeriesServiceDatabaseException e) {
-                    logger.error("Error reading properties of series {}", seriesId, e);
-                  }
-
-                  // do the actual index update
-                  updateSeriesInIndex(seriesId, organization.getId(),
-                          updateFunctions.toArray(new Function[0]));
-
-                });
-        logIndexRebuildProgress(logger, index.getIndexName(), total, current);
-        current++;
+
+                    // do the actual index update
+                    updateSeriesBundleInIndex(seriesIds, organization.getId(),
+                            updateFunctionBundle);
+
+                  });
+          for (int i = 1; i <= seriesBundle.size(); i++) {
+            int logEntry = i + (current - seriesBundle.size());
+            logIndexRebuildProgress(logger, index.getIndexName(), total, logEntry);
+          }
```

Comment:
I wouldn't generate a bunch of logs simultaneously, but just log the current state instead.